app/src/main/java/org/oppia/android/app/home/HomeViewModel.kt

```diff
@@ -129,15 +154,29 @@ class HomeViewModel(
    * ongoing stories at all.
    */
   private fun computePromotedStoryViewModelList(
-    ongoingStoryList: OngoingStoryList
+    recommendedStoryList: RecommendedStoryList
   ): List<PromotedStoryViewModel> {
-    val storyList = if (ongoingStoryList.recentStoryCount != 0) {
-      ongoingStoryList.recentStoryList
-    } else {
-      // TODO(#936): Optimise this as part of recommended stories.
-      ongoingStoryList.olderStoryList
+    val storyList = when {
+      recommendedStoryList.suggestedStoryCount != 0 -> {
+        val list = if (recommendedStoryList.recentlyPlayedStoryCount != 0 ||
+          recommendedStoryList.olderPlayedStoryCount != 0
+        ) {
+          recommendedStoryList.recentlyPlayedStoryList +
+            recommendedStoryList.olderPlayedStoryList +
+            recommendedStoryList.suggestedStoryList
+        } else {
+          recommendedStoryList.suggestedStoryList
+        }
+        list
```

Comment:
doesn't look correct to me. 
we can skip the use of `val list = ` as if and else will return `List<PromotedStory>`
Correct me if I am wrong here on anything.

Reply:
Yes you are right I can skip this. I just tried replacing things and thought it would look more simplified. Removed variable and used if else.